### PR TITLE
Fixes for location widget

### PR DESCRIPTION
**Pull request by nop33**
Remaining issues:
* [x] https://github.com/running-coder/jquery-typeahead/issues/192
* [ ] https://github.com/running-coder/jquery-typeahead/issues/193

Test cases: https://gist.github.com/nop33/ca26c574cad390328a620acd9b2a85f2

Visual issues in IE:
* [x] No highlighting
* [x] Placeholder looks like text

### CODE REVIEW

**Comment by pferreir** [comment]
screen shot 2016-05-23 at 10 33 45

File: indico/web/templates/forms/location_widget.html [hunk 1]
```diff
@@ -448,10 +455,13 @@
                         this.query = cleanQuery(query);
                     },
                     onShowLayout: function() {
+                        console.log(this.node.attr('id') + ': onShowLayout')
+                        toggleOpenResultsList(this.node);
                         roomInput.siblings('.keyword-highlighter').removeClass('visible');
                         currentScrollPos = 0;
                     },
                     onHideLayout: function(node) {
+                        console.log(node.attr('id') + ': onHideLayout')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage2:

Comment (ThiefMaster) [inline indico/web/templates/forms/location_widget.html]: :monkey:

File: indico/web/templates/forms/location_widget.html [hunk 2]
```diff
@@ -479,6 +491,7 @@
             });
 
             venueInput.on('typeahead:click-on-option', function(e, data) {
+                console.log(venueInput.attr('id') + ': typeahead:click-on-option')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage4:

File: indico/web/templates/forms/location_widget.html [hunk 3]
```diff
@@ -491,6 +504,7 @@
                 venueInput.attr('title', data.item_name);
             })
             .on('typeahead:click-on-custom-option', function(e, data) {
+                console.log(venueInput.attr('id') + ': typeahead:click-on-custom-option')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage1:

File: indico/web/templates/forms/location_widget.html [hunk 4]
```diff
@@ -500,39 +514,54 @@
                 venueInput.val(data.item_name);
                 validVenue = true;
                 venueInput.attr('title', data.item_name);
+                venueInput.focus();
                 venueInput.trigger('typeahead:close-results-list');
             });
 
-            roomInput.on('typeahead:click-on-custom-option', function(e, data) {
+            roomInput.on('typeahead:click-on-option', function(e, data) {
+                console.log(roomInput.attr('id') + ': typeahead:click-on-option')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage3:

File: indico/web/templates/forms/location_widget.html [hunk 5]
```diff
@@ -500,39 +514,54 @@
...
+                updateVenueOnRoomChange(data);
+                hiddenData.room_id = data.id;
+                delete hiddenData.room_name;
+                postSelectionActions(roomInput, data.name);
+                highlightOption(roomInput, data.name);
+                roomInput.attr('title', data.name);
+            }).on('typeahead:click-on-custom-option', function(e, data) {
+                console.log(roomInput.attr('id') + ': typeahead:click-on-custom-option')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage1:

File: indico/web/templates/forms/location_widget.html [hunk 6]
```diff
@@ -500,39 +514,54 @@
...
                 postSelectionActions(roomInput, data.item_name);
                 roomInput.val(data.item_name);
                 validRoom = true;
                 roomInput.attr('title', data.item_name);
+                roomInput.focus();
                 roomInput.trigger('typeahead:close-results-list');
             });
 
             $('#{{ field.id }}-wrapper .i-location-input-field').on('typeahead:close-results-list', function() {
+                console.log(this.id + ': typeahead:close-results-list')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage1:

File: indico/web/templates/forms/location_widget.html [hunk 7]
```diff
@@ -500,39 +514,54 @@
...
                 global.Typeahead["#" + this.id].hideLayout();
             }).on('typeahead:open-results-list', function() {
+                console.log(this.id + ': typeahead:open-results-list')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage1:

File: indico/web/templates/forms/location_widget.html [hunk 8]
```diff
@@ -500,39 +514,54 @@
...
                 global.Typeahead["#" + this.id].showLayout();
             })
             .on('focus.typeahead', function() {
                 /* In case the input is cleared, this ensures that the results dropdown will contain all available
                 options. */
                 var $this = $(this);
-                if (roomJustCleared && $this.is(roomInput)) {
+                console.log($this[0].id + ': focus.typeahead')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage4:

File: indico/web/templates/forms/location_widget.html [hunk 9]
```diff
@@ -448,10 +455,13 @@
                         this.query = cleanQuery(query);
                     },
                     onShowLayout: function() {
+                        console.log(this.node.attr('id') + ': onShowLayout')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage1:

File: indico/web/templates/forms/location_widget.html [hunk 10]
```diff
@@ -384,10 +390,13 @@
...
+                        toggleOpenResultsList(this.node);
                         venueInput.siblings('.keyword-highlighter').removeClass('visible');
                         currentScrollPos = 0;
                     },
                     onHideLayout: function(node) {
+                        console.log(node.attr('id') + ': onHideLayout')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage3:

File: indico/web/templates/forms/location_widget.html [hunk 11]
```diff
@@ -384,10 +390,13 @@
                         this.query = cleanQuery(query);
                     },
                     onShowLayout: function() {
+                        console.log(this.node.attr('id') + ': onShowLayout')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage4:

File: indico/web/templates/forms/location_widget.html [hunk 12]
```diff
@@ -318,18 +319,23 @@
             }
 
             function handleKeystrokes(node) {
+                console.log(node.attr('id') + ': handleKeystrokes')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage1:

File: indico/web/templates/forms/location_widget.html [hunk 13]
```diff
@@ -290,25 +293,23 @@
...
-                } else if (inputField.is(roomInput) && venueInput.closest('.typeahead__container').hasClass('result')) {
-                    venueInput.trigger('typeahead:close-results-list');
-                    roomInput.trigger('typeahead:close-results-list');
-                    roomInput.trigger('typeahead:open-results-list');
                 }
             }
 
             function resetSelectedOption() {
-                if (latestUsedField.is(venueInput)) {
+                console.log(latestUsedField[0].id + ': resetSelectedOption')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage2:

File: indico/web/templates/forms/location_widget.html [hunk 14]
```diff
@@ -290,25 +293,23 @@
             }
 
             function toggleOpenResultsList(inputField) {
-                if (inputField.is(venueInput) && roomInput.closest('.typeahead__container').hasClass('result')) {
+                console.log(inputField[0].id + ': toggleOpenResultsList')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage2:

File: indico/web/templates/forms/location_widget.html [hunk 15]
```diff
@@ -175,9 +176,11 @@
             function updateVenueOnRoomChange(item) {
                 /* Override the venue based on the room selection */
                 if (item.venue_id != hiddenData.venue_id) {
+                    console.log('updateVenueOnRoomChange')
```
Comment (pferreir) [inline indico/web/templates/forms/location_widget.html]: :rage1:

**Comment by pferreir** [comment]
Merged in 353f952.

**Pull request closed without merging**